### ISSUE TITLE
Touch screen initialization failed..

### ISSUE THREAD

**Issue by Boyeen** [mjdonders/CST816_TouchLib#10]
No matter what I do, different examples, Arduino IDE/PlatformIO I can't get the touch system to init.
All examples do the same.

I'm able to use my TTG T-DisplayS3 and the LilyGo demos (especially touch). I have implemented the LilyGo touch library in my code but of course it is a very blunt instrument :-) so I was excited to see your lib.

Any suggestions what I might be doing wrong?

**Comment by mjdonders** [comment]
Assuming you have the touch screen version of this device: did you do the Wire.begin before this library begin method?
(that's the mistake I've made myself more than once)

I can check your code if you share it?
The examples should all work without any change. 

Oh, and another option: did you check the serial port speed of your IDE?
As-in: can you see any message from the device?

Hoping this helps.
Please let me know

**Comment by Boyeen** [comment]
Thanks so much for your reply (and your work on the lib).  Yes, I have the touch version but I am wondering now if they changed touch chips? Mine is over a year old and probably from the earliest batches.

I have been trying all your examples with zero changes, so yes, the wire.begin is there. Also, I see the "Initialization failed" message so I am getting messages. Very strange.  I will keep trying  and let you know if I am successful. Again, you replied in zero time, thanks so much :-)

**Comment by mjdonders** [comment]
[test_print_device_info.zip](https://github.com/mjdonders/CST816_TouchLib/files/13538405/test_print_device_info.zip)
You could try this.
It initializes the touchscreen (but does not check the result) and print the device type (again: with little error handling here).
I just tested it, with a T-Display S3 I had here. 
It outputs 'Device is of type: CST816T'. 
Maybe your board has another type of touch chip?

**Comment by mjdonders** [comment]
Oh, one last remark: please try without connected to anything else, in order to rule out any (faulty connected) PCB.

**Comment by Boyeen** [comment]
Thanks again for your time on this. Here is my output:


10:29:30.813 -> Touch screen initialization failed..
10:29:30.813 -> Device is of type: Unknow device type: 0
10:29:30.813 -> Halting program.


So, I went back and dug up some code I wrote a year ago, a simple drawing program and I believe I found the problem.
I based my code on a LilyGo example which uses “TouchLib" by Micky. According to that library constructor
my TTGO-S3-touch uses a different touch controller:  CTS328
Could be the same family and someone swapped the TS for ST…. but it is not compatible with your library anyway.

It also explains why I have not been able to make any other touch library examples for the TTGO-S3-touch work :-)
I may have a weird early model.

So, my apologies for wasting your time. I will continue to try to get some basic gestures working for my 
current project.  Feel free to close this issue and you have been extremely kind with your time to help me.
This is the spirit of Arduino :-)

Thanks again.

Tim

**Comment by Boyeen** [comment]
Thanks again for your time on this. Here is my output:


10:29:30.813 -> Touch screen initialization failed..
10:29:30.813 -> Device is of type: Unknow device type: 0
10:29:30.813 -> Halting program.


So, I went back and dug around some more.  First, my LilyGo docs have SCL/SDA mixed up from yours, I have tried both ways with no luck.

Second, according to the Touchlib by Micky, my TTGO-S3-touch uses a different touch controller:  CTS328. Somebody got their letters mixed up I think but anyway.

I ran an I2C address scan and the only device on my S3 is at 1A. I tried modifying your lib
to use 1A but still no luck.

At this point I am totally confused :-)


Thanks again for your help.

Tim
